lib/monitoring/aws-rds/RdsInstanceMetricFactory.ts

```diff
@@ -0,0 +1,139 @@
+import { DimensionsMap } from "aws-cdk-lib/aws-cloudwatch";
+import { IDatabaseInstance } from "aws-cdk-lib/aws-rds";
+
+import {
+  LatencyType,
+  MetricFactory,
+  MetricStatistic,
+  getLatencyTypeLabel,
+  getLatencyTypeStatistic,
+} from "../../common";
+
+const RdsNamespace = "AWS/RDS";
+
+export interface RdsInstanceMetricFactoryProps {
+  /**
+   * database instance
+   */
+  readonly instance?: IDatabaseInstance;
+}
+
+export class RdsInstanceMetricFactory {
+  readonly instanceIdentifier: string;
+  readonly instance?: IDatabaseInstance;
+  protected readonly metricFactory: MetricFactory;
+  protected readonly dimensionsMap: DimensionsMap;
+
+  constructor(
+    metricFactory: MetricFactory,
+    props: RdsInstanceMetricFactoryProps
+  ) {
+    this.metricFactory = metricFactory;
+    this.instance = props.instance;
+    this.instanceIdentifier =
+      RdsInstanceMetricFactory.resolveDbInstanceIdentifier(props);
+    this.dimensionsMap = { DBInstanceIdentifier: this.instanceIdentifier };
+  }
+
+  private static resolveDbInstanceIdentifier(
+    props: RdsInstanceMetricFactoryProps
+  ): string {
+    if (!props.instance) {
+      throw Error("`instance` is required");
+    }
+    return props.instance.instanceIdentifier;
+  }
+
+  metricTotalConnectionCount() {
+    return this.metric(
+      "DatabaseConnections",
+      MetricStatistic.SUM,
+      "Connections: Sum"
+    );
+  }
+
+  metricAverageCpuUsageInPercent() {
+    return this.metric("CPUUtilization", MetricStatistic.AVERAGE, "CPU Usage");
```

Comment:
We could just use the helper functions like https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.IDatabaseInstance.html#metricwbrcpuutilizationprops similar to [this](https://github.com/cdklabs/cdk-monitoring-constructs/blob/f59da8e8061d625eb7ae9b622512479a715a146f/lib/monitoring/aws-sns/SnsTopicMetricFactory.ts#L19-L22).